src/client/opamPinCommand.ml

```diff
@@ -764,10 +764,26 @@ let list st ~short =
                   (OpamPackage.version_to_string inst))]
         with Not_found -> OpamConsole.colorise `yellow "(uninstalled)", []
       in
+      let vcs_revision = 
+        let url = OpamStd.Option.default OpamFile.URL.empty url in
+        match kind with 
+        | "git" | "hg" | "darcs" ->
```

Comment:
You can match on `OpamUrl.backend` here. Here is an example:
https://github.com/ocaml/opam/blob/8e3ee689a9cf34aa0c6203f8d948b2d20075d9e5/src/state/opamFileTools.ml#L394-L397